File: frontend/src/microservices/advertising/objects/ListAdvertisementsRequest.ts
// 本文件定义 advertising 模块的 `ListAdvertisementsRequest`，用于列表查询入参并提供 JSON 编解码。

export type ListAdvertisementsRequest = {
  placement?: string | null
  reviewStatus?: string | null
  reviewStatuses?: string[] | null
  ownerManagerId?: string | null
  ownerType?: string | null
  deliverableOnly?: boolean | null
  currentTime?: string | null
}

export const listAdvertisementsRequestFromJson = (json: string): ListAdvertisementsRequest =>
  JSON.parse(json) as ListAdvertisementsRequest

export const listAdvertisementsRequestToJson = (value: ListAdvertisementsRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/ListAdvertisementsRequest.scala
// 本文件定义 advertising 模块的 ListAdvertisementsRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ListAdvertisementsRequest(
    placement: Option[String],
    reviewStatus: Option[String],
    reviewStatuses: Option[List[String]],
    ownerManagerId: Option[String],
    ownerType: Option[String],
    deliverableOnly: Option[Boolean],
    currentTime: Option[String]
)

object ListAdvertisementsRequest:
  given sourceEncoder: Encoder[ListAdvertisementsRequest] = deriveEncoder
  given sourceDecoder: Decoder[ListAdvertisementsRequest] = deriveDecoder

